frontends/concrete-python/concrete/fhe/mlir/converter.py

```diff
@@ -57,46 +57,76 @@ def convert(
             MlirModule:
                 In-memory MLIR module corresponding to the graph
         """
-
-        self.process(graph)
+        self.process(graphs)
 
         with mlir_context as context, MlirLocation.unknown():
             concrete.lang.register_dialects(context)  # pylint: disable=no-member
 
             module = MlirModule.create()
             with MlirInsertionPoint(module.body):
-                ctx = Context(context, graph, self.configuration)
+                for name, graph in graphs.items():
+                    # pylint: disable=cell-var-from-loop
+                    # ruff: noqa: B023
+                    ctx = Context(context, graph, self.configuration)
+
+                    input_types = [ctx.typeof(node).mlir for node in graph.ordered_inputs()]
+
+                    @func.FuncOp.from_py_func(*input_types, name=name)
+                    def main(*args):
+                        for index, node in enumerate(graph.ordered_inputs()):
+                            conversion = Conversion(node, args[index])
+                            if "original_bit_width" in node.properties:
+                                conversion.set_original_bit_width(
+                                    node.properties["original_bit_width"]
+                                )
+                            ctx.conversions[node] = conversion
+
+                        ordered_nodes = [
+                            node
+                            for node in nx.lexicographical_topological_sort(graph.graph)
+                            if node.operation != Operation.Input
+                        ]
+
+                        for progress_index, node in enumerate(ordered_nodes):
+                            self.trace_progress(self.configuration, progress_index, ordered_nodes)
+                            preds = [ctx.conversions[pred] for pred in graph.ordered_preds_of(node)]
+                            self.node(ctx, node, preds)
+                        self.trace_progress(self.configuration, len(ordered_nodes), ordered_nodes)
+
+                        outputs = []
+                        for node in graph.ordered_outputs():
+                            assert node in ctx.conversions
+                            outputs.append(ctx.conversions[node].result)
+
+                        return tuple(outputs)
 
-                input_types = [ctx.typeof(node).mlir for node in graph.ordered_inputs()]
+        return module
 
-                @func.FuncOp.from_py_func(*input_types)
-                def main(*args):
-                    for index, node in enumerate(graph.ordered_inputs()):
-                        conversion = Conversion(node, args[index])
-                        if "original_bit_width" in node.properties:
-                            conversion.set_original_bit_width(node.properties["original_bit_width"])
-                        ctx.conversions[node] = conversion
+    def convert(
```

Comment:
If this is remaining, why are the tests changed?

Reply:
Tests are changed because they use the `process` method. Not this one.